Dev 2.0.0 to master

CraZyRc opened the pull request.

GnarLito committed:
Refactor Game / Translator

GnarLito committed:
Added code coverage
fixed translator and game

GnarLito committed:
Fixed showing version warning when running a to new version.

GnarLito committed:
converted onGridHook for cooldown to Walk on/off hook

GnarLito committed:
imported SQLite

GnarLito committed:
Created beginning for DB
created GameDB class

CraZyRc committed:
v2.0.0 - ALPHA

stuff being made!

CraZyRc committed:
v2.0.0 - ALPHA

stuff being made!
Stuff being shit

CraZyRc committed:
v2.0.0 - ALPHA

stuff being made!
Stuff being fixed

GnarLito committed:
Fixed PlayerOnMove crash

GnarLito committed:
Make end of cooldown more snappy

GnarLito committed:
Fixed Folder creation

GnarLito committed:
Moved Player Yeet to game runner

GnarLito committed:
Fixed AlltranslationsPressent test

GnarLito committed:
Temporarily removed Translation Manager.AlreadyActive

CraZyRc committed:
v2.0.0 - ALPHA

Language fix

CraZyRc committed:
Actionbar fix
extra messages in Translator
(Full inventory, restart after cd done)

CraZyRc committed:
added item modifier in config
added streak concept in game
added invalidhammeritem text in translator
added moleshit and hitstreak in database

CraZyRc committed:
scoreboard lists and /top command

GnarLito committed:
Added save gamefile header saving

CraZyRc committed:
mole heads configurable per game in settings.

fixed the reset ticket not working
fixed the actionbar messages
fixed the gamefile header

added messages for the head skin commands

CraZyRc committed:
Merge branch 'dev_2.0.0' of https://github.com/CraZyRc/WhackaMole into dev_2.0.0

# Conflicts:
#	src/main/java/whackamole/whackamole/Game.java

CraZyRc committed:
1.17 gamefile header fix

GnarLito committed:
Table generics (#8)

* Unsued code cleanup

* New DB Model implementation

* Furter implementation Table generics

* Replaced Translator Form check with assertion

* Added Insert() return row id

* Added IsUnique option to Column

* Better SQL Error Logs
* Table Column validation

* Converted Cooldown Table To new implemtation

* Updated pom + small translation test fix

* Translation Format fix / CooldownDB column mismatch Fix

* Moved Testing setup to own file

* Converted Game DB / row to new class

* Added boolean type / safe query value casting

* Changed return type for Insert from void to type T

* Added Cooldown DB insert Test

* Addded SQL defaults for Columns

* Converted ScoreBoard DB to new class / added tests

* Converted gridDB to new class

* Converted Game Settings class to extend GameRow class

* General changes

* Added scoreStreak row, Changed game score to scroeboard row

* Implemented Delete query for all tables

* Restructured table.java

* Added row sub class convertion

* Added select and delete from DB

* Add documentation to GameRow

* Fixed game file changes to DB

* review fixes

* score streak fix

GnarLito committed:
Fix broken merge from TableGenerics

CraZyRc committed:
top score message almost fully fixed.

added new font size system, now only misses spaces for 1 and 2 pix width

GnarLito committed:
fixed score not saving correctly

CraZyRc committed:
added teleportation command for intruders on the game field.

added the start for a visible scoreboard system

added positions command

CraZyRc committed:
Merge branch 'dev_2.0.0' of https://github.com/CraZyRc/WhackaMole into dev_2.0.0

# Conflicts:
#	src/main/java/whackamole/whackamole/Game.java

GnarLito committed:
Implemented Bukkit.Location to SQL

GnarLito committed:
Added ISerializer generics

GnarLito committed:
Converted All Serializers to Generic types

CraZyRc committed:
Commands:
- adjusted minor teleport changes for location

Game:
- added kill on game remove
- changed the holo spawn methods
- added a holo updater

GameDB:
- fixed the locations from separate X,Y,Z to 1 location save

GamesManager:
- fixed minor bug that didn't show the NOGAMESFOUND message when no games exist.

CraZyRc committed:
Merge branch 'dev_2.0.0' of https://github.com/CraZyRc/WhackaMole into dev_2.0.0

# Conflicts:
#	src/main/java/whackamole/whackamole/DB/GameDB.java
#	src/main/java/whackamole/whackamole/DB/GameRow.java

CraZyRc committed:
Implemented SQL Date serializer

CraZyRc committed:
Commands:
- added the streak command
- fixed the top command with date

Game:
- added streakholo location to gameconfig
- changed player to offlineplayer
- added holograms for game top scores and currentstreak
- added logic for current streak counting

GameDB:
- added streakhologram location to database

Translator:
- added streakholo location explanation for gameconfig

CraZyRc committed:
GamesManager:
- made changes for games folder generation

YMLFile:
- added games folder generation

GnarLito committed:
general cleanup

GnarLito committed:
Added plugin disable for when the Config does not want to load

GnarLito committed:
Fixed typo in config

GnarLito committed:
Added Config validation tests

GnarLito commented on src/main/java/whackamole/whackamole/Main.java:
```diff
@@ -1,48 +1,48 @@
-        Main.plugin = this;
+
+        Logger.onLoad(this);
+        if (!Config.onLoad(this)) {
```
@CraZyRc If u agree to handle the config loading failing this way, will u then add a translation for this?

GnarLito committed:
fixed creating file issues creating folders.

GnarLito committed:
Yml file fix (#9)

YML file fixed

GnarLito committed:
Added Save loading from DB and files at the same time

CraZyRc committed:
Game:
- changed file deletion so it won't log the message all the time
- changed world name in load from grid to settings

GamesManager:
- removed bug where world was null but still called upon

YMLFFile:
- made sure `created` turns true upon file creation
